### ISSUE TITLE
tikz matrix entry separator conflicts with side-by-side in LaTeX

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#156]
Reporter suggests that the ampersand used in a tikz matrix conflicts with the `\pushValignCaptionBottom` macro.  Example code: plain is problem-free, side-by-side fails.

https://groups.google.com/forum/#!topic/mathbook-xml-support/Bqd87biU83Y

```xml
<figure>
<image xml:id="image1">
<latex-image-code>
<![CDATA[

\begin{tikzpicture}
\matrix
{
  a & b\\
};
\end{tikzpicture}
]]>
</latex-image-code>
</image>
</figure>

<sidebyside>
<figure>
<image xml:id="image2">
<latex-image-code>
<![CDATA[

\begin{tikzpicture}
\matrix
{
  a & b\\
};
\end{tikzpicture}
]]>
</latex-image-code>
</image>
</figure>
```

**Comment by cmhughes** [comment]
I'll experiment and report back.

Initial thoughts: it's weird that this causes issues, but that tabulars do
not.

**Comment by rbeezer** [comment]
Thanks, Chris.  Yes, quite a corner case.  As I said on the support list, it'd be best if an author did not have to markup a matrix differently just for tikz inside a side-by-side.

So hopefully we can *configure* a fix.

Rob

**Comment by cmhughes** [comment]
Before I look at this, did it turn out to be an xelatex issue?

**Comment by Alex-Jordan** [comment]
This issue is really more of a `tikz` issue with `&` than a `sidebyside` issue, but I won't go down the rabbit hole of explaining. It means Chris is off the hook, and it's on me somewhat to come up with ideas since I'm responsible for the whole `latex-image-preamble` scheme.

The current standard is to use `\amp` inside tex code in MBX source, so rather than `\&` that A Pereira offered in the Google group post that started this, the solution should assume the author uses `\amp`.

Despite `\amp` having been defined in the preamble, it's not recognized within the `\matrix` command. I don't know why not, but that's beside the point. What is needed is the line:

``\tikzset{ampersand replacement = \amp}``

in the `latex-image-preamble` (which will be used in both tex output and for making image files). This should be used whenever the `tikz` package is loaded in someone's `latex-image-preamble`. Unfortunately, that's not something you can 100% automatically test for. Because `tikz` might be in play as a dependency of some other package the author put in their `latex-image-preamble`.

So I propose the XSL search `latex-image-preamble` for "tikz", "pgplots", and a handful of other common packages that use `tikz`. If any of these are found, the above line gets thrown in right after the latex image preamble. For more obscure packages that use `tikz` (or in any case) the author could be expected to manually include the above line in their `latex-image-preamble`.

This proposal involves:

- some XSL around `latex-image-preamble` in `mathbook-latex.xsl` and `extract-latex-image.xsl`
- requiring authors use \amp (not `&`, not `\&`) inside tex code in MBX source
- in rare cases, asking authors to include the line `\tikzset{ampersand replacement = \amp}` in their `latex-image-preamble`.

**Comment by rbeezer** [comment]
This should be fixed at #257.
